### ISSUE TITLE
tf.device is not working on tflearn.DNN model

### ISSUE THREAD

**Issue by jaesik817** [tflearn/tflearn#761]
Hi, 

I have used tflearn to quick prototyping, and I found I set device as cpu for parallel running, and the models were using gpu memory.

I captured part of my code in here.

  with tf.device('/cpu:0'):
    # Network building
    net = tflearn.input_data([None, FLAGS.input_length,len(FLAGS.select_list.split(","))])
    net = tflearn.lstm(net, len(FLAGS.select_list.split(","))*FLAGS.lstm_num);
    net = tflearn.fully_connected(net, len(FLAGS.select_list.split(","))*FLAGS.output_length);
    net = tflearn.regression(net, optimizer='sgd', learning_rate=0.1,metric='R2',
                         loss='mean_square');
    # Training
    model = tflearn.DNN(net, tensorboard_verbose=0);

Thanks.

**Comment by aymericdamien** [comment]
Yes, TensorFlow will always allocate all available GPU memory, to avoid it you can run:
`export CUDA_VISIBLE_DEVICES=''`

**Comment by jaesik817** [comment]
Hm, I checked that one, however after setting CUDA_VISIBLE_DEVICES as "", I found the model tried to use gpu memory.
Fri May 19 10:38:01 2017
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 375.26                 Driver Version: 375.26                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1080    Off  | 0000:81:00.0     Off |                  N/A |
|  8%   48C    P2    45W / 230W |   7720MiB /  8113MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 1080    Off  | 0000:C1:00.0     Off |                  N/A |
|  7%   47C    P2    45W / 230W |   7715MiB /  8113MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0    113407    C   python                                        7718MiB |
|    1    113407    C   python                                        7713MiB |
+-----------------------------------------------------------------------------+

i338425@seltera46:/home2/i338425/work/tensorflow/pathnet> echo $CUDA_VISIBLE_DEVICES

Could I know there are other reasons? Thanks.

**Comment by jaesik817** [comment]
I solved that by changing CUDA_VISIBLE_DEVICES value.
